### DISCUSSION TITLE
Ruby 3.2: YJIT for all

### DISCUSSION THREAD

**Discussion by FooBarWidget** [fullstaq-labs/fullstaq-ruby-server-edition#116] [Show and tell]
Thanks to the contributions of @san983 (#115), we now provide Ruby 3.2 packages. What's more, YJIT is enabled.

Enabling YJIT deserves special mention because it wasn't so easy. Compiling YJIT requires Rust edition 2021 support, but not all distributions provide a recent enough Rust package. In particular Debian doesn't — not even Debian 11! We decided against disabling YJIT on Debian, and went through the effort to install a recent enough Rust compiler for all distributions that we target.

Enjoy Ruby 3.2 with YJIT support!

**Comment by san983** [comment]
Again, thanks @FooBarWidget for your guidance!

**Comment by JlordA** [comment]
This is an old thread but hoping someone here is still listening but I am building new server images for my job and using fullstaq-ruby for 3.2.3 binaries.  I am wondering do the fullstaq-ruby binaries for 3.2.3 have the YJIT enabled by default or are there additional steps I would need to make in my build and deploy scripts? @FooBarWidget and @san983

**Reply by Envek** [reply]
The same as with “vanilla” Ruby you need to enable YJIT explicitly in runtime via either `--yjit` CLI option or `RUBY_YJIT_ENABLE` environment variable set to `1` or in your code.

See https://github.com/ruby/ruby/blob/master/doc/yjit/yjit.md#usage for all options.

E.g. for [containerized Fullstaq Ruby](https://github.com/evilmartians/fullstaq-ruby-docker):

```sh
docker run -it --rm -e RUBY_YJIT_ENABLE=1 quay.io/evl.ms/fullstaq-ruby:3.3-jemalloc-slim ruby --version
```

Output should contain `+yjit`
